src/tracker.jl

```diff
@@ -0,0 +1,34 @@
+abstract type Tracker end
+
+function init_tracker(tracker::Tracker, image::Array{T, 2}, bounding_box::MVector{4, Int}) where T
+    @assert bounding_box[1] >= 1 && bounding_box[1] <= size(image)[1]
+    @assert bounding_box[2] >= 1 && bounding_box[2] <= size(image)[2]
+    @assert bounding_box[3] >= 1 && bounding_box[3] <= size(image)[1]
+    @assert bounding_box[4] >= 1 && bounding_box[4] <= size(image)[2]
+
+    @assert bounding_box[1] < bounding_box[3]
+    @assert bounding_box[2] < bounding_box[4]
+
+    init_impl(tracker, image, bounding_box)
```

Comment:
is `*_impl` a placeholder, or you'll implement your code in this way?
Perhaps you could just implement all your code in a single `init_tracker` function; but with multiple methods.
We have some discussion about the `*_impl` style here: https://github.com/zygmuntszpak/ImageBinarization.jl/issues/23#issuecomment-471506862

Reply:
> is `*_impl` a placeholder, or you'll implement your code in this way?

My intention of using `*_impl` is:
While object-tracking, one can initialise a tracker as `tr = TrackerAlgoName` and then call `init_tracker()` and `update_tracker()` invariantly for all the trackers they want to use. 
And have different `*_impl` methods for each of the trackers whose implementation can be found in separate sub-folders for each tracker. This will ensure code clearity.


> Perhaps you could just implement all your code in a single `init_tracker` function; but with multiple methods.

I don't think there is much of difference in what I have done and what you are trying to say. I have just put those `init_tracker` under a single umbrella.
Please guide me if I am wrong.

Reply:
I'm not very clear the implementation details, but If you're going to define some `Trakcer` type, and if you're going to do things like this:

```julia
abstract type Tracker end
struct TrackerAlgoName_A <: Tracker end
struct TrackerAlgoName_B <: Tracker end
```

then you could implement `init_tracker` using constructors `TrackerAlgoName_A(image)`, which is clearer.

and the implementation function is:
```julia
update_tracker(tracker::TrackerAlgoName_A, image)
update_tracker(tracker::TrackerAlgoName_B, image)
```

Reply:
Before you implement the code details, I'd like to ask one more questions:

Is there a real need for `init_tracker` rather than a Constructor? Or in other words, is there any case we need a not-yet-initialized `Tracker`?

Reply:
> Is there a real need for `init_tracker` rather than a Constructor? Or in other words, is there any case we need a not-yet-initialized `Tracker`?

No, may be not. I can't think off a situation for that.
I can initialize tracker in the constructor itself.

Reply:
> Is there a real need for `init_tracker` rather than a Constructor? Or in other words, is there any case we need a not-yet-initialized `Tracker`?

I think that a Constructor simply initialises the fields in the struct/class but the `init_tracker` method here does much more than that. 

The trackers implemented here follow an online learning process where they learn the representation of the object and background at each frame in the video sequence. They get better at identifying the object from the background as they get trained for more and more frames (`update_tracker`). What the `init_tracker` method does is that it runs the learning process on the initial image for a large number of iterations so that the tracker is trained enough so as to be able to identify the object in the very next frame (i.e. when `update_tracker` is called for the first time). All this is not usually done in a conventional constructor so I had created a separate method for this.

Reply:
Oops. I didn't see @Deepank308's reply below :sweat_smile:. Really sorry